website_sale_coupon_restrict/models/sale_coupon.py

```diff
@@ -4,9 +4,10 @@
 
 
 class SaleCoupon(models.Model):
-    _inherit = "sale.coupon"
+    _inherit = "coupon.coupon"
 
-    def _check_coupon_code(self, order):
+    def _check_coupon_code(self, order_date, partner_id, **kwargs):
+        order = kwargs.get("order", False)
         if self.program_id.website_only and not order.website_id:
```

Comment:
You should check if order exists as well

Reply:
Done, thanks for the suggest